Enable AOTI shim v2 build and add into libtorch

hl475 opened the pull request:
Summary:
Follow up of https://github.com/pytorch/pytorch/pull/125087

This diff will create shim v2 header and cpp file and corresponding build

Differential Revision: D56617546

hl475 commented:
put this on hold due to the extra comment in https://github.com/pytorch/pytorch/pull/120513

desertfire commented:
As we discussed offline, https://github.com/pytorch/pytorch/pull/125589 switched to generate only .cpp files at the build time. Header files are stored in PyTorch repo.

hl475 committed:
Enable AOTI shim v2 build and add into libtorch (#125211)

Summary:

Follow up of https://github.com/pytorch/pytorch/pull/125087

This diff will create shim v2 header and cpp file and corresponding build

Differential Revision: D56617546

hl475 commented:
@pytorchbot merge
